model/src/main/java/com/chiralbehaviors/CoRE/Ruleform.java

```diff
@@ -104,6 +153,199 @@
         return entity;
     }
 
+    @SuppressWarnings("unchecked")
+    public static <T extends Ruleform> T sanitize(T ruleform) throws IOException {
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+        serialize(ruleform, os);
+        ByteArrayInputStream is = new ByteArrayInputStream(os.toByteArray());
+        return (T) deserialize(is, ruleform.getClass());
+    }
+
+    public static void serialize(Ruleform ruleform,
+                                 OutputStream os) throws IOException {
+        ObjectMapper objMapper = new ObjectMapper();
+        objMapper.registerModule(new CoREModule());
+        Hibernate4Module module = new Hibernate4Module();
+        module.enable(Feature.FORCE_LAZY_LOADING);
+        objMapper.registerModule(module);
+        objMapper.writerWithDefaultPrettyPrinter()
+                 .writeValue(os, ruleform);
+    }
+
+    public static Map<UUID, Ruleform> slice(Ruleform ruleform,
+                                            Predicate<Ruleform> systemDefinition,
+                                            Map<UUID, Ruleform> sliced,
+                                            Set<UUID> traversed) {
+        map(ruleform, systemDefinition, sliced, traversed);
+        return sliced;
+    }
+
+    @SuppressWarnings("unchecked")
+    public static <T extends Ruleform> T smartMerge(EntityManager em,
+                                                    T ruleform,
+                                                    Map<UUID, Ruleform> mapped) {
+        return (T) map(em, ruleform, mapped);
+    }
+
+    public static void traverse(Ruleform ruleform,
```

Comment:
refactored to normalize as much as I could about the traversal logic.  cleaned up logic.